plone/namedfile/utils.py

```diff
@@ -77,3 +87,240 @@ def stream_data(file):
+        # next 2 Bytes always Number: 42
+        content_type = 'image/tiff'
+        w = -1
+        h = -1
+        # check for '42' as flag for tiff:
+        # TODO: implement correct Image Length and Image Width lookup --> Page 14ff
+        # Page 18: Tags:
+        # ImageLength: Tag: 257 (101.H) Short or Long
+        # ImageWidth: Tag: 256 (100.H) Short or Long
+        try:
+            if data[:2] == 'MM' and struct.unpack('>H', data[2:4])[0] == 42:
+                log.info("Tiff Image in big-endian encoding")
+                # big-endian encoding for the whole data stream
+                offset = 7 + struct.unpack('>HH', data[4:8])[0]
+                while offset < size and (w == -1 or h == -1):
+                    tag = struct.unpack('>H', data[offset:offset + 2])[0]
+                    stype = struct.unpack('>H', data[offset + 2:offset + 4])[0]
+                    value = struct.unpack('>HH', data[offset + 4:offset + 8])[0]
+                    new_offset = struct.unpack('>HH', data[offset + 8:offset + 12])[0]
+                    if tag == 256:  # tag 256: ImageWidth (100.H) Short or Long
+                        w = value
+                    elif tag == 257:  # tag 257: ImageLength (101.H) Short or Long
+                        h = value
+                    log.info("Found Tag: %s at Offset: %s; Type: %s; Value: %s; new Offset: %s",
+                             tag, offset, stype, value, new_offset)
+                    offset = offset + new_offset
+            elif data[:2] == 'II' and struct.unpack('<H', data[2:4])[0] == 42:
+                log.info("Tiff Image in little-endian encoding")
+                # litle-endian
+                offset = 7 + struct.unpack('<HH', data[4:8])[0]
+                while offset < size and (w == -1 or h == -1):
+                    tag = struct.unpack('<H', data[offset:offset + 2])[0]
+                    stype = struct.unpack('<H', data[offset + 2:offset + 4])[0]
+                    value = struct.unpack('<HH', data[offset + 4:offset + 8])[0]
+                    new_offset = struct.unpack('<HH', data[offset + 8:offset + 12])[0]
+                    if tag == 256:  # tag 256: ImageWidth (100.H) Short or Long
+                        w = value
+                    elif tag == 257:  # tag 257: ImageLength (101.H) Short or Long
+                        h = value
+                    log.info("Found Tag: %s at Offset: %s; Type: %s; Value: %s; new Offset: %s",
+                             tag, offset, stype, value, new_offset)
+                    offset = offset + new_offset
+            else:
+                # not a tiff image
+                pass
+            width = int(w)
+            height = int(h)
+        except struct.error:
+            pass
+        except ValueError:
+            pass
+        except TypeError:
+            pass
+    # Use PIL / Pillow to determ Image Information
+    else:
+        img = PIL.Image.open(StringIO(data))
```

Comment:
Some test are currently failing at this point with error messages like: `cannot identify image file <StringIO.StringIO instance at 0x7f8a58ff8ab8>`
As a short-cut, if the image is empty, you could probably return immediately if size is 0 at the beginning of this method.

And as an additional safeguard, wrap this call to `PIL.Image.open` into a try-except for IOError. This is the error that gets raised in PIL.Image.open:

    raise IOError("cannot identify image file %r"
                  % (filename if filename else fp))